### PR TITLE
Adding Faster R-CNN guide

### PR DESCRIPTION
**Pull request by kapil-varshney**
This adds the Deep Learning - Faster R-CNN guide.

### CODE REVIEW

File: guide/14-deep-learning/faster-rcnn-object-detector.ipynb [hunk 1]
```diff
@@ -0,0 +1,190 @@
+{
```
Comment (vikashranjan26) [inline guide/14-deep-learning/faster-rcnn-object-detector.ipynb:1]: <p>As we discussed, Please fix the last sentence.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/Esri/arcgis-python-api/pull/897/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='14'/>

**Review by vikashranjan26 — APPROVED** [review]
LGTM

**Review by priyankatuteja — APPROVED** [review]

**Comment by priyankatuteja** [comment]
As discussed please make the following minor changes related to paper citation. @BP-Ent Could you do the review too.

File: guide/14-deep-learning/faster-rcnn-object-detector.ipynb (hunk 1 above)
Comment (BP-Ent) [inline guide/14-deep-learning/faster-rcnn-object-detector.ipynb:1]: <p>We have seen how the one-shot object detection models, such as <a href="https://developers.arcgis.com/python/guide/how-ssd-works/" target="_blank">SSD</a>, <a href="https://developers.arcgis.com/python/guide/how-retinanet-works/" target="_blank">RetinaNet</a>, and YOLOv3, work. However, before the single-stage detectors were the norm, the most popular object detectors were from the multi-stage R-CNN family. First, there was R-CNN, then Fast R-CNN came along with some improvements, and then eventually, Faster R-CNN became the state-of-the-art multi-stage object detector. As is obvious from the names, these models evolved from one to the next, providing better performance at faster speeds. Although Faster R-CNN is not as fast as some of the later single-stage models, it remains one of the most accurate object detection models.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/Esri/arcgis-python-api/pull/897/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='1'/>

Comment (BP-Ent) [inline guide/14-deep-learning/faster-rcnn-object-detector.ipynb:1]: <p>Until Faster R-CNN came out, its contemporaries were using various algorithms for <s>the </s>region proposal <s>part which</s> <em>that</em> were being computed on the CPU and <s>becoming the</s> <em>creating a</em> bottleneck. Faster R-CNN improved the object detection architecture by replacing the selection search algorithm in Fast R-CNN with a convolutional network called the Region Proposal Network (RPN). The rest of the model architecture remains the same as Fast R-CNN - the image is fed to a CNN to produce a feature map from which features for regions proposed by the RPN are selected and resized by a pooling layer and fed to an FC layer with two heads, a softmax classifier, and a bounding box regressor. This design <s>speeded up</s><em> increased the speed of</em> detection and brought it closer to real-time [1].</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/Esri/arcgis-python-api/pull/897/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='11'/>

Comment (BP-Ent) [inline guide/14-deep-learning/faster-rcnn-object-detector.ipynb:1]: <p>Region Proposal Network, like other region proposal algorithms, inputs an image and returns regions of interest that contain objects. An RPN also returns an "objectness" score <s>which</s> <em>that</em> measures how likely the region is<s> supposed</s> to have an object vs. a background [1]. In Faster R-CNN, the RPN and the detect network share the same backbone. The last shared layer of the backbone provides a feature map of the image <s>which</s> <em>that</em> is used by the RPN to propose regions.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/Esri/arcgis-python-api/pull/897/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='14'/>

Comment (BP-Ent) [inline guide/14-deep-learning/faster-rcnn-object-detector.ipynb:1]: <p>Using a sliding window approach, a small network is <s>used over</s> <em>overlaid onto </em>the feature map. At each spatial window, there are multiple anchor boxes that are of predetermined scales and aspect ratios. This enables the model to detect objects of a wide range of scales and aspect ratios in the same image. Usually, three different scales and three different aspect ratios are used, resulting in nine anchor boxes at each spatial location<s>, which</s> <em>that </em>denote<s>s</s> the maximum region proposals at that spatial location. The small network then feeds into fully connected layers with two heads - one for the "objectness" score and the other for the bounding box coordinates of the region proposals. Figure 5 demonstrates the sliding window at a spatial location, the corresponding anchor boxes, and how <s>it</s> <em>they </em>feed<s>s</s> into the subsequent layers [1]. Note that these layers, though similar to the last few layers of the Fast R-CNN object detector, are not the same, nor do they share the <em>same </em>weights. The RPN classifies the regions in a class agnostic manner, as it is tasked with only finding the regions <s>which</s> <em>that </em>contain the objects. </p><p>In this guide, we have tried to present an intuition to understand the Faster R-CNN model. To dive into more details about the model and it's architecture, please read the original papers mentioned in <em>the </em>references below. You can <em>also </em>reference this <a href="https://towardsdatascience.com/faster-r-cnn-for-object-detection-a-technical-summary-474c5b857b46" target="_blank">blog</a>.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/Esri/arcgis-python-api/pull/897/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='16'/>

**Review by BP-Ent — CHANGES_REQUESTED** [review]
Suggested changes made on ReviewNB

File: guide/14-deep-learning/faster-rcnn-object-detector.ipynb (hunk 1 above)
Comment (AtmaMani) [inline guide/14-deep-learning/faster-rcnn-object-detector.ipynb:1]: <p>Replace title to read the title of the page. We need this so the browser tab, SE indexers get them correctly</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/Esri/arcgis-python-api/pull/897/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='0'/>

**Review by AtmaMani — CHANGES_REQUESTED** [review]
requested update to title

File: guide/14-deep-learning/faster-rcnn-object-detector.ipynb (hunk 1 above)
Comment (kapil-varshney) [reply guide/14-deep-learning/faster-rcnn-object-detector.ipynb:1]: Added a title to the guide with the same name as the notebook file. Does this help?

Comment (AtmaMani) [reply guide/14-deep-learning/faster-rcnn-object-detector.ipynb:1]: <p>yes, thanks</p>

**Review by AtmaMani — APPROVED** [review]

**Pull request merged**